src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -233,33 +233,44 @@ private void OnAsyncStartupCompleted(StartupEventArgs e)
             && CommandLineParser.TryParse<WebWindowOptions>(e.Args, out var webWindowOptions)
             && webWindowOptions.Url != null)
         {
-            var moduleId = Guid.NewGuid().ToString();
-
-            var moduleCatalog = _host!.Services.GetRequiredService<ModuleCatalog>();
-            moduleCatalog.Add(new WebModuleManifest
-            {
-                Id = moduleId,
-                Name = webWindowOptions.Url,
-                ModuleType = ModuleType.Web,
-                Details = new WebManifestDetails
-                {
-                    Url = new Uri(webWindowOptions.Url),
-                    IconUrl = webWindowOptions.IconUrl == null ? null : new Uri(webWindowOptions.IconUrl)
-                }
-            });
-
-            var moduleLoader = _host.Services.GetRequiredService<IModuleLoader>();
-            moduleLoader.StartModule(new StartRequest(moduleId, new List<KeyValuePair<string, string>>
-            {
-                new(WebWindowOptions.ParameterName, JsonSerializer.Serialize(webWindowOptions))
-            }));
+            CreateAndStartNewWebModuleAsync(
```

Comment:
Is this intentionally not awaited?

Reply:
I assume so. this is still just the same functionality with the organization into a new method.